rocAL/source/api/rocal_api_augmentation.cpp

```diff
@@ -2680,3 +2679,180 @@ rocalNonLinearBlendFixed(
     }
     return output;
 }
+
+RocalTensor ROCAL_API_CALL
+rocalDilate(RocalContext p_context,
+            RocalTensor p_input,
+            bool is_output,
+            unsigned kernel_size,
+            RocalTensorLayout output_layout,
+            RocalTensorOutputType output_datatype) {
+    Tensor* output = nullptr;
+    ROCAL_INVALID_CONTEXT_ERR(p_context, output);
+    ROCAL_INVALID_INPUT_ERR(p_input, output);
+    auto context = static_cast<Context*>(p_context);
+    auto input   = static_cast<Tensor*>(p_input);
+    try {
+        RocalTensorlayout op_tensor_layout  = static_cast<RocalTensorlayout>(output_layout);
+        RocalTensorDataType op_tensor_dtype = static_cast<RocalTensorDataType>(output_datatype);
+        TensorInfo output_info = input->info();
+        output_info.set_tensor_layout(op_tensor_layout);
+        output_info.set_data_type(op_tensor_dtype);
+        output = context->master_graph->create_tensor(output_info, is_output);
+        context->master_graph->add_node<DilateNode>({input}, {output})->init(kernel_size);
```

Comment:
Non-blocking: Should we sanity-check `kernel_size` before we hand it to RPP? The kernels expect a positive, typically odd size; letting 0 or an even value through from the API is likely to result in a runtime error later. Failing fast here with a clear message would make the Python/C++ surface a lot friendlier.